src/diffusers/pipelines/fabric/pipeline_fabric.py

```diff
@@ -0,0 +1,552 @@
+            new_config = dict(unet.config)
+            new_config["sample_size"] = 64
+            unet._internal_dict = FrozenDict(new_config)
+
+        self.register_modules(
+            unet=unet,
+            vae=vae,
+            text_encoder=text_encoder,
+            tokenizer=tokenizer,
+            scheduler=scheduler,
+        )
+        self.vae_scale_factor = 2 ** (len(self.vae.config.block_out_channels) - 1)
+        self.image_processor = VaeImageProcessor(vae_scale_factor=self.vae_scale_factor)
+
+    def initialize_prompts(self, prompts: List[str], device):
+        # Breaking into individual prompts feels memory efficient
+        prompt_embed_list = []
+        for prompt in prompts:
+            prompt_tokens = self.tokenizer(
+                prompt,
+                return_tensors="pt",
+                max_length=self.tokenizer.model_max_length,
+                padding="max_length",
+                truncation=True,
+            )
+
+            attention_mask = (
+                prompt_tokens.attention_mask.to(device)
+                if (
+                    hasattr(self.text_encoder.config, "use_attention_mask")
+                    and self.text_encoder.config.use_attention_mask
+                )
+                else None
+            )
+
+            prompt_embd = self.text_encoder(
+                input_ids=prompt_tokens.input_ids.to(device),
+                attention_mask=attention_mask,
+            ).last_hidden_state
+
+            prompt_embed_list.append(prompt_embd)
+
+        all_prompt_embed = torch.cat(prompt_embed_list, dim=0)
+        return all_prompt_embed
+
+    def get_unet_hidden_states(self, z_all, t, prompt_embd):
+        cached_hidden_states = []
+        for module in self.unet.modules():
+            if isinstance(module, BasicTransformerBlock):
+
+                def new_forward(self, hidden_states, *args, **kwargs):
+                    cached_hidden_states.append(hidden_states.clone().detach().cpu())
+                    return self.old_forward(hidden_states, *args, **kwargs)
+
+                module.attn1.old_forward = module.attn1.forward
+                module.attn1.forward = new_forward.__get__(module.attn1)
+
+        # run forward pass to cache hidden states, output can be discarded
+        _ = self.unet(z_all, t, encoder_hidden_states=prompt_embd)
+
+        # restore original forward pass
+        for module in self.unet.modules():
+            if isinstance(module, BasicTransformerBlock):
+                module.attn1.forward = module.attn1.old_forward
+                del module.attn1.old_forward
+
+        return cached_hidden_states
+
+    def unet_forward_with_cached_hidden_states(
+        self,
+        z_all,
+        t,
+        prompt_embd,
+        cached_pos_hiddens: Optional[List[torch.Tensor]] = None,
+        cached_neg_hiddens: Optional[List[torch.Tensor]] = None,
+        pos_weights=(0.8, 0.8),
+        neg_weights=(0.5, 0.5),
```

Comment:
Should these be registered as configuration variables in the pipeline? IIUC these should be exposed to the users for better controllability?

Reply:
make sense